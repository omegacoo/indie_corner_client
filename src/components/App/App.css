.App {
    color: white;
    max-width: 60vw;
    margin-left: auto;
    margin-right: auto;
}