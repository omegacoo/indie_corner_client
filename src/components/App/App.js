import React from 'react';
import { Route } from 'react-router-dom';

import Header from '../Header/Header';
import MainLanding from '../MainLanding/MainLanding';
import StoreContext from '../../StoreContext';

import './App.css';

export default class App extends React.Component {

    render(){
        let contextValue = {};
        return(
            <div className='App'>
                <StoreContext.Provider value={contextValue}>
                    {/* HEADER COMPONENT */}
                    <Route 
                        component={Header}
                    />

                    {/* MAIN COMPONENT */}
                    <Route 
                        exact path={'/'}
                        component={MainLanding}
                    />
                </StoreContext.Provider>
            </div>
        );
    };
};