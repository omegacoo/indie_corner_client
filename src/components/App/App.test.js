import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import App from './App';

describe('App component', () => {
    it('renders without crashing', () => {
        const div = document.createElement('div');

        ReactDOM.render(
            <BrowserRouter>
                <App />
            </BrowserRouter>, div);
        ReactDOM.unmountComponentAtNode(div);
    })

    it('renders UI correctly', () => {
        expect(
            renderer.create(
                <BrowserRouter>
                    <App />
                </BrowserRouter>).toJSON()
        ).toMatchSnapshot()
    })
})