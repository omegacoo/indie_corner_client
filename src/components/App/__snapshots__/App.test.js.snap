// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders UI correctly 1`] = `
<div
  className="App"
>
  <div
    className="Header"
  >
    <h1
      className="Header-title"
    >
      Indie Corner
    </h1>
  </div>
  <div
    className="MainLanding"
  >
    <h2
      className="MainLanding-title"
    >
      Welcome to THE forum for the Indie Gaming Community!
    </h2>
    <button
      className="MainLanding-login-button"
    >
      Log in
    </button>
    <button
      className="MainLanding-register-button"
    >
      Register
    </button>
    <button
      className="MainLanding-anonymous-button"
    >
      Go Anonymous
    </button>
  </div>
</div>
`;
